### ISSUE TITLE
Byte-compiler warns about usage of obsolete generalized variable `point`

### ISSUE THREAD

**Issue by meliache** [alphapapa/hammy.el#9]
When byte-compiling hammy.el on the latest emacs master branch, I get the warning
```
hammy.el:681:18: Warning: ‘point’ is an obsolete generalized variable; use
    ‘goto-char’ instead.
```

It has been made obsolete in emacs 29, see
Emacs 29 [NEWS](https://github.com/emacs-mirror/emacs/blob/84963956535e9eff38b0d7638ce751039eb44bc2/etc/NEWS#L3787-L3811). 

The warning comes from the following lines:
https://github.com/alphapapa/hammy.el/blob/f1f38d811cfd4d8fdf528c7cabeb62a52cc07a2f/hammy.el#L675-L681

Obviously you are aware of `goto-char`, as it's used a couple of lines above already with `(goto-char (point-max))`. Thefore I don't understand why we need `(setf (point) (point-max))` _again_, and without understanding that, I didn't want to make a PR. Is it just because we just inserted something? Seems like we could just use this `goto-char` again same as above and for the `buffer-window` case just something like `(with-current-buffer buffer-window (goto-char (point-max))` or something like that, no?

**Comment by alphapapa** [comment]
I probably used it because it parallels the use of `(setf (window-point...` in the THEN clause.

I wish that `point` had not been marked obsolete as a generalized variable.  It's a useful paradigm, and no justification was given for its marking, so it seems like a matter of preference.  I posted about it on emacs-devel a while back.  I'll probably leave these few uses of it in my packages, despite the byte-compiler's complaining.

**Comment by meliache** [comment]
Okay, as long it's not deprecated and the package works I can live with a warning, so fine with me :man_shrugging: It's just that if I see the warning in a couple of packages then it's probably not as seldomly used as is claimed in the NEWS file. I just grepped my packages for `setf (point` and that's used in hammy and ement by you, and elfeed and emacs-web-server. I'm not sure if there are language-design considerations why this was decided. For better or worse, in Emacs there have always been many ways to achieve something.

I joined the emacs-devel mailinglist only this year, so probably have missed that thread. Can I somehow reply to a thread before I joined? If not, I can just start a new thread.

**Comment by alphapapa** [comment]
> Okay, as long it's not deprecated and the package works I can live with a warning, so fine with me

Right, it should only happen when the package is being installed and compiled, not every time it's loaded or used.  (And most Elisp packages cause a lot of such warnings--which is unfortunate, but it's not entirely in the package developers' control, because a new version of Emacs can suddenly start producing new warnings.)

> It's just that if I see the warning in a couple of packages then it's probably not as seldomly used as is claimed in the NEWS file. I just grepped my packages for setf (point and that's used in hammy and ement by you, and elfeed and emacs-web-server.

Yes, it seems to be more widely used than some of the Emacs maintainers thought.

> I'm not sure if there are language-design considerations why this was decided. 

AFAICT from the git commit and list discussion, it was a rather whimsical decision, not researched ahead of time, nor discussed publicly.

> For better or worse, in Emacs there have always been many ways to achieve something.

Yes, and more are being added (see the many new, low-level, context-specific functions in Emacs 29 as an example), so I see no reason to get rid of these generalized variable setters.

> I joined the emacs-devel mailinglist only this year, so probably have missed that thread. Can I somehow reply to a thread before I joined? If not, I can just start a new thread.

There are a few ways I know of:

1.  You could use Gnus with the GMANE NNTP server to find old threads and post to them.
2. You could download the relevant monthly file from the mailing list Web server, open it in a mail client, and start a reply from the appropriate message.
3. You could find the appropriate message ID (e.g. from one of the public-inbox-based frontends to the list) and write a new message, adding the message ID to the appropriate email headers to mark it as a reply.
4.  Just start a new thread and include a link to the old one.

If you do, please let me know (and please cc me).  I won't promise to chime in on it, but I might, if I have time.  With all the to-dos I already have related to Emacs, I haven't felt like picking up the torch for this issue, especially the way I was asked to individually defend each one I might want to use, rather than putting the burden where it properly belongs, on the one who wants to remove existing functionality.

Thanks.
